src/validation/permission-exclusions.ts

```diff
@@ -0,0 +1,303 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files
+ are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Vijaya Kumar Guthi <vijaya.guthi@modusbox.com>
+ --------------
+ ******/
+
+import * as keto from '@ory/keto-client'
+import { ValidationError } from './validation-error'
+import { ServiceConfig } from '../shared/config'
+import { logger } from '../shared/logger'
+
+export interface UserRole {
+  username: string;
+  roles: string[];
+}
+
+export interface RolePermissions {
+  rolename: string;
+  permissions: string[];
+}
+
+export interface PermissionExclusions {
+  permissionsA: string[];
+  permissionsB: string[];
+}
+
+export interface PermissionExclusionCombos {
+  permissionA: string;
+  permissionB: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function isPermissionExclusionCombos (object: PermissionExclusions | PermissionExclusionCombos): boolean {
+  return 'permissionA' in object
+}
```

Comment:
This form of the function means you no longer need to cast after you use it:
```suggestion
export function isPermissionExclusionCombos (
  o: PermissionExclusions[] | PermissionExclusionCombos[]
): o is PermissionExclusionCombos[] {
  return o.length !== 0 && 'permissionA' in o[0]
}
```

Reply:
Also note that I don't think you need to export this function at all.